pinot-core/src/test/java/org/apache/pinot/core/data/function/DateTimeFunctionsTest.java

```diff
@@ -778,4 +787,14 @@ public void testDateTimeConvertMultipleInvocations() {
     testMultipleInvocations(String.format("dateTimeConvert(timeCol, '%s', '%s', '%s')", inputFormatStr, outputFormatStr,
         outputGranularityStr), rows, expectedResults);
   }
+
+  @Test
+  public void testSleepFunction() {
+    long startTime = System.currentTimeMillis();
+    testFunction("sleep(500)", Collections.emptyList(), new GenericRow(), result -> {
+      assertTrue((long) result >= 500);
+    });
+    long endTime = System.currentTimeMillis();
+    assertTrue(endTime - startTime >= 500);
+  }
```

Comment:
nit: we can reduce time to something in the order of tens of millis

Reply:
50 ms, final offer

b458a5b286